urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/DefaultResponseBuilder.java

```diff
@@ -0,0 +1,398 @@
+package ikrs.httpd;
+
+/**
+ * @author Ikaros Kappler
+ * @date 2012-07-16
+ * @version 1.0.0
+ **/
+
+import java.io.IOException;
+import java.net.Socket;
+import java.util.MissingResourceException;
+import java.util.UUID;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.logging.Level;
+
+import ikrs.httpd.resource.ByteArrayResource;
+import ikrs.httpd.response.GeneralPreparedResponse;
+import ikrs.httpd.response.successful.OK;
+
+import ikrs.typesystem.*;
+
+
+public class DefaultResponseBuilder
+    extends AbstractResponseBuilder
+    implements ResponseBuilder {
+
+    
+    /**
+     * For the case the request cannot be processed it will be forwarded to the internal
+     * error response builder.
+     **/
+    private ErrorResponseBuilder errorResponseBuilder;
+
+
+    /**
+     * The constructor.
+     **/
+    public DefaultResponseBuilder( HTTPHandler handler ) {
+	super( handler );	
+
+	this.errorResponseBuilder = new ErrorResponseBuilder(handler);
+    }
+
+    /**
+     * The default response builder uses a nested response builder for error processing.
+     * In some special cases it can be helpful to access the ErrorResponseBuilder directy (such as
+     * the HTTPHandler does if the threaded execution is rejected by the ThreadPoolExecutor).
+     *
+     * This method simply returns the internal error response builder.
+     *
+     * @return The internal error response builder.
+     **/
+    protected ErrorResponseBuilder getErrorResponseBuilder() {
+	return this.errorResponseBuilder;
+    }
+
+
+    //---BEGIN------------------------------ ResponseBuilder implementaion -----------------------
+    /**
+     * This method translates the given headers and socket into an executable 
+     * PreparedResponse object.
+     *
+     * The method does not throw any exceptions as the error reporting is part of HTTP
+     * itself.
+     *
+     * @param headers  The previously processed headers.
+     * @param postData The actual sent non-header data (if available; may be null).
+     * @param socketID The unique socket ID.
+     * @param socket   The acutual socket.
+     * @param additionals A map containing non-essential builder params. The expected 
+     *                    map contents depends on the underlying implementation; some
+     *                    builders even allow null-additionals.
+     *
+     * @return A new HTTPRequest built from the HTTP headers.
+     *
+     **/
+    public PreparedHTTPResponse create( HTTPHeaders headers,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.